rosbag2_transport/test/rosbag2_transport/test_play.cpp

```diff
@@ -1244,8 +1245,14 @@ TEST_F(RosBag2PlayQosOverrideTestFixture, topic_qos_profiles_overridden_incompat
     topic_name_, num_msgs_to_wait_for_, qos_request);
   play_options_.topic_qos_profile_overrides = topic_qos_profile_overrides;
 
+  // QoS incompatibilities are not expected with rmw_zenoh_cpp.
+  bool expect_timeout = true;
+  if (std::strcmp(rmw_get_implementation_identifier(), "rmw_zenoh_cpp") == 0) {
```

Comment:
what about using here `qos_check_compatible` ? In other test we do somthing like:

```cpp
  if (qos_check_compatible(qos_profile_publisher,
    qos_profile_subscription).compatibility != rclcpp::QoSCompatibility::Ok)
  {
```

Reply:
Sorry, I have a bit opposed opinion. If `rmw_zenoh` do not expect any QoS incompatibilities it doesn't make sense to run this test at all.   
In my opinion, it would be much cleaner if we would skip this test for the `rmw_zenoh` with an explanation like 
```
// If the RMW implementation is rmw_zenoh_cpp, we do not expect any QoS incompatibilities.
```
rather than trying to work around expectations in this test.

Reply:
IMO, incompatible or not, it needs to call the API to the rmw implementation since that is not what rosbag2 or clients interface can decide. in the case of other rmw implementation support or deprecating rmw implementation in the future, we do not need to come back here to maintain. i think, that is better for the maintenance and integrity of the test. but i am okay with @MichaelOrlov opinion, if anybody agrees with that 👍